### PR TITLE
Translate Eval Command

### LINKED ISSUE #429: [FEATURE]Add `eval` PPL command
**Is your feature request related to a problem?**
Add `eval` command for PPL spark based execution driver

- [PPL eval command](https://github.com/opensearch-project/sql/blob/d3ecb2d75712db2bcbd2945e62728c6599580657/docs/user/ppl/cmd/eval.rst)

**Do you have any additional context?**
 - https://github.com/opensearch-project/sql/blob/d3ecb2d75712db2bcbd2945e62728c6599580657/docs/user/ppl/cmd/eval.rst
 - https://github.com/opensearch-project/opensearch-spark/issues/408

### PR DESCRIPTION
**Pull request by LantaoJin**
### Description
This PR translates the PPL `eval` command to Spark LogicalPlan by adding a Project with Alias expressions.

### Issues Resolved
Resolves #429 

### Check List
- [x] Updated documentation (ppl-spark-integration/README.md)
- [x] Implemented unit tests
- [x] Implemented tests for combination with other commands
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ast/expression/Field.java [hunk 1]
```diff
@@ -8,25 +8,25 @@
 import com.google.common.collect.ImmutableList;
 import org.opensearch.sql.ast.AbstractNodeVisitor;
 
-import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+
 public class Field extends UnresolvedExpression {
-  private final UnresolvedExpression field;
+  private final QualifiedName field;
```
Comment (YANG-DB) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ast/expression/Field.java:15]: are we always assuming the field will be a QualifiedName ? 
are there cases it wont be able to resolve ?

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java [hunk 2]
```diff
@@ -172,20 +172,6 @@ public UnresolvedExpression visitPercentileAggFunction(OpenSearchPPLParser.Perce
                 Collections.singletonList(new Argument("rank", (Literal) visit(ctx.value))));
     }
 
-    @Override
```
Comment (YANG-DB) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java:175]: why was this removed?

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ast/expression/Field.java (hunk 1 above)
Comment (LantaoJin) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ast/expression/Field.java:15]: Yes. This ensured by PPL syntax. https://github.com/opensearch-project/opensearch-spark/blob/main/ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4#L327

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java (hunk 2 above)
Comment (LantaoJin) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java:175]: It was removed in other PR which merged in main branch already. Since I started this work for a while so the code base in this PR has out of date I guess. So I manually removed this in my code base to avoid compile failure. Will merge main branch later.

Comment (LantaoJin) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java]: After merging with main branch, this git diff is gone.

**Review by dai-chen — APPROVED** [review]

**Review by YANG-DB — APPROVED** [review]
